### ISSUE TITLE
[FEA] local bridges (similar to Nx)

### ISSUE THREAD

**Issue by geoHeil** [rapidsai/cugraph#1418]
Improve Networkx API compatibility: add support for local bridges:

https://networkx.org/documentation/stable//reference/algorithms/generated/networkx.algorithms.bridges.local_bridges.html#networkx.algorithms.bridges.local_bridges

**Comment by BradReesWork** [comment]
@geoHeil thanks for the suggestion, we will investigate level of effort and work this request into the backlog

**Comment by geoHeil** [comment]
FYI: networkx works well when computing the local bridges without the shortest paths (with_span=False) https://stackoverflow.com/questions/66291151/how-to-consume-a-python-gneerator-in-parallel-using-multiprocessing/66302419#66302419

When trying to replace networkx with cugraph in a naive way i.e. only substituting the shortest path computation it still is really slow - as the whole loop is still running in python on a single core.

**Comment by BradReesWork** [comment]
dropping the compatibility portion of this issues since that is captured in another issues and ties into the on-going work with NetworkX.   The request for local bridges is still valid

**Comment by ChuckHastings** [comment]
local_bridges can be constructed using the technique from edge triangle counting, which is being added in 24.06.  Instead of counting, we can remove any vertex pairs that are part of a triangle.

After #4382 is merged into the 24.06 release we should be able to schedule this work when we have someone to implement it.

**Comment by ChuckHastings** [comment]
One of our oldest open issues.

We have a new package nx-cugraph that is focused on our NX integration.  We can certainly explore adding this to the nx-cugraph package (and providing whatever support is require within libcugraph to support this).

Trying to identify the use cases for this to help assign a priority.  The nx-cugraph team hasn't had requests for this algorithm to be migrated onto the GPU yet.  Is this still important to you?  If so, what is your use case and how critical do you see this being?

**Comment by geoHeil** [comment]
It is not important anymore for me.
